docs/CLI/datasets.md

```diff
@@ -64,22 +130,286 @@ $ primehub datasets list
+```
+
+### Admin actions for datasets
+
+These actions only can be used by administrators:
+
+* create
+* update
+* delete
+
+For `create` and `update` require a dataset configuration, please see above examples.
+
+### Fields for creating or updating
+
+| field | required | type | description |
+| --- | --- | --- | --- |
+| name | required | string | it should be a valid resource name for kubernetes |
+| displayName | optional | string | display name for this dataset |
+| description | optional | string | |
+| global | optional | boolean | when a dataset is global, it could be seen for each group |
+| type | required | string | one of ['pv', 'nfs', 'hostPath', 'git', 'env'] |
+| url | conditional | string | **MUST** use with `git` type |
+| pvProvisioning | conditional | string | onf of ['auto', 'manual'], **MUST** use with `pv` type. This field only uses in `CREATE` action |
+| nfsServer | conditional | string | **MUST** use with `nfs` type |
+| nfsPath | conditional | string | **MUST** use with `nfs` type |
+| hostPath | conditional | string | **MUST** use with `hostPath` type  |
+| variables | optional | dict | **MAY** use with `env` type. It is key value pairs. All values have to a string value. For example: `{"key1":"value1","key2":"value2"}`. |
+| groups | optional | list of connected groups (dict) | please see the `connect` examples |
+| secret | optional | dict | **MAY** use with `git` type | bind a `secret` to the `git` dataset |
+| volumeSize | conditional | integer | **MUST** use with `pv` type. The unit is `GB`.|
+| enableUploadServer | optional | boolean | it only works with one of ['pv', 'nfs', 'hostPath'] writable types |
+
+> There is a simple rule to use fields for `UPDATE`. All required fields should not be in the payload.
+
+For example, there is a configuration for creating env dataset:
+
+```bash
+primehub datasets create <<EOF
+{
+  "name": "env-dataset",
+  "description": "",
+  "type": "env",
+  "variables": {
+    "ENV": "prod",
+    "LUCKY_NUMBER": "7"
+  }
+}
+EOF
+```
+
+After removing required `name` and `type` fields, it could be used with updating:
+
+```bash
+primehub datasets update env-dataset <<EOF
+{
+  "description": "make changes to the description",
+  "variables": {
+    "ENV": "prod",
+    "LUCKY_NUMBER": "8"
+  }
+}
+EOF
+```
+
+For updating, giving things that you want to make different:
+
+```bash
+primehub datasets update env-dataset <<EOF
+{
+    "groups": {
+      "connect": [
+        {
+          "id": "a7a283b5-c0e2-4b79-a78c-39c630324762",
+          "writable": false
+        }
+      ]
+    }
+  }
+EOF
+```
+
+
+
+
+
+### PV type
+
+```json
+{
+  "name": "pv-dataset",
+  "displayName": "the dataset created by SDK",
+  "description": "It is a PV dataset",
+  "type": "pv",
+  "global": false,
+  "groups": {
```

Comment:
Could we remove the connection in the PV type example? or users would fail to use this example because group id is not found.